### ISSUE TITLE
Failed to parse source map from [...]/providers/transactions/reducer.ts

### ISSUE THREAD

**Issue by clement4saunier** [TrueFiEng/useDApp#680]
```
WARNING in ./node_modules/@usedapp/core/dist/esm/src/providers/transactions/reducer.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '[project]/node_modules/@usedapp/core/src/providers/transactions/reducer.ts' file: Error: ENOENT: no such file or directory, open '[project]/node_modules/@usedapp/core/src/providers/transactions/reducer.ts'
 @ ./node_modules/@usedapp/core/dist/esm/src/providers/transactions/provider.js 8:0-47 23:46-64
 @ ./node_modules/@usedapp/core/dist/esm/src/providers/DAppProvider.js 8:0-62 67:31-50
 @ ./node_modules/@usedapp/core/dist/esm/src/providers/index.js 1:0-31 1:0-31
 @ ./node_modules/@usedapp/core/dist/esm/src/index.js 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149
 @ ./src/index.js 8:0-63 12:19-34 14:5-20 15:5-20 21:33-45
```

I get this error whenever I use any version >0.7.0. It's a warning so it does compile, but it then causes a lot of issues like unexpectedly reverted calls even though they work using the same provider, and many many invalid hash errors that bugs out my project.

![image](https://user-images.githubusercontent.com/47207681/162712214-975f3c66-4d53-49aa-836c-959a2760e2ba.png)

Using webpack 5.72.0.

**Comment by dmaretskyi** [comment]
The warning should be fixed if we include sources

**Comment by dmaretskyi** [comment]
Not exactly sure about invalid hash, are you passing undefined where a transaction hash is expected? Can you find a piece of code that triggers that kind of error?

**Comment by clement4saunier** [comment]
No i don't pass an undefined transaction hash, that's the thing. Even when I don't call any hooks I get this error so I figured it comes from the Dapp Provider, but I could be wrong.

**Comment by chand1012** [comment]
I'm having the same issue on `@usedapp/core` versions 1.0.0 and 0.12.9. Using CRA Typescript template.

```
WARNING in ./node_modules/@usedapp/core/dist/esm/src/providers/transactions/context.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from './node_modules/@usedapp/core/src/providers/transactions/context.ts' file: Error: ENOENT: no such file or directory, open './node_modules/@usedapp/core/src/providers/transactions/context.ts'
 @ ./node_modules/@usedapp/core/dist/esm/src/providers/index.js 6:0-64 6:0-64
 @ ./node_modules/@usedapp/core/dist/esm/src/index.js 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149
 @ ./src/index.tsx 9:0-45 16:33-45

WARNING in ./node_modules/@usedapp/core/dist/esm/src/providers/transactions/model.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from './node_modules/@usedapp/core/src/providers/transactions/model.ts' file: Error: ENOENT: no such file or directory, open './node_modules/@usedapp/core/src/providers/transactions/model.ts'
 @ ./node_modules/@usedapp/core/dist/esm/src/providers/index.js 8:0-37 8:0-37
 @ ./node_modules/@usedapp/core/dist/esm/src/index.js 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149
 @ ./src/index.tsx 9:0-45 16:33-45

WARNING in ./node_modules/@usedapp/core/dist/esm/src/providers/transactions/provider.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from './node_modules/@usedapp/core/src/providers/transactions/provider.tsx' file: Error: ENOENT: no such file or directory, open './node_modules/@usedapp/core/src/providers/transactions/provider.tsx'
 @ ./node_modules/@usedapp/core/dist/esm/src/providers/DAppProvider.js 8:0-62 67:31-50
 @ ./node_modules/@usedapp/core/dist/esm/src/providers/index.js 1:0-31 1:0-31
 @ ./node_modules/@usedapp/core/dist/esm/src/index.js 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149
 @ ./src/index.tsx 9:0-45 16:33-45

WARNING in ./node_modules/@usedapp/core/dist/esm/src/providers/transactions/reducer.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from './node_modules/@usedapp/core/src/providers/transactions/reducer.ts' file: Error: ENOENT: no such file or directory, open './node_modules/@usedapp/core/src/providers/transactions/reducer.ts'
 @ ./node_modules/@usedapp/core/dist/esm/src/providers/transactions/provider.js 8:0-47 23:46-64
 @ ./node_modules/@usedapp/core/dist/esm/src/providers/DAppProvider.js 8:0-62 67:31-50
 @ ./node_modules/@usedapp/core/dist/esm/src/providers/index.js 1:0-31 1:0-31
 @ ./node_modules/@usedapp/core/dist/esm/src/index.js 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149 3:0-149
 @ ./src/index.tsx 9:0-45 16:33-45
```

It looks like including the source code in your distribution would be an easy, albeit temporary, fix for this issue while a better solution is came up with.

**Comment by FanManutd** [comment]
Same problem as @chand1012

**Comment by noodesonmars** [comment]
same issue here, seems to be preventing useEthersBalance or any hook from working, makes entire package faulty

**Comment by noodesonmars** [comment]
I fixed it by downgrading useDapp to `^0.11.0`, react to '^17.0.2`, and typescript to `4.1.2`, for anyone that needs a quick fix. I'm not sure if each downgrade is necessary though.

**Comment by noodesonmars** [comment]
> same issue here, seems to be preventing useEthersBalance or any hook from working, makes entire package faulty, `yarn create react-app projectName --template typescript`, `react: ^18.0.0`, `@usedapp/core: ^1.0.0`

I fixed it by downgrading useDapp to `^0.11.0`, react to `^17.0.2`, and typescript to `4.1.2`, for anyone that needs a quick fix. I'm not sure if each downgrade is necessary though.

**Comment by rzadp** [comment]
This should be working properly now.
I have followed the steps with `yarn create react-app`, adding latest `@usedapp/core` and `ethers`, and now the problem seems to be gone.

There is a similar warning:

```
Failed to parse source map from 'xxx/node_modules/@metamask/detect-provider/src/index.ts' file: Error: ENOENT: no such file or directory, open 'xxx/node_modules/@metamask/detect-provider/src/index.ts'
```

But this is not a problem in our code.
